AzureIotPnpDps.X/platform/cli/cli.c

```diff
@@ -309,11 +341,186 @@ static void get_cli_version(char *pArg)
+           "---------------------"NEWLINE \
+           "LED Number"NEWLINE \
+           "1 = Blue"NEWLINE \
+           "2 = Green"NEWLINE \
+           "3 = Yellow"NEWLINE \
+           "4 = Red"NEWLINE \
+           "---------------------"NEWLINE \
+           "LED State"NEWLINE \
+           "1 = On"NEWLINE \
+           "2 = Off"NEWLINE \
+           "3 = Fast Blink"NEWLINE \
+           "4 = Slow Blink"NEWLINE \
+           NEWLINE );
+    return;
+}
+
+static void set_led(char *pArg)
+{
+    //	LED number
+    //	1 = Blue
+    //	2 = Green
+    //	3 = Yellow
+    //	4 = Red
+    //
+    //	State
+    //	1 = On
+    //	2 = Off
+    //	3 = Blink
+    char *led_param[3];
+    char *pch;
+    uint8_t i;
+    uint8_t params = 0;
+    uint8_t led_num = 0;
+    uint8_t led_state = 0;
+    led_set_state_t new_led_state = LED_STATE_OFF;
+
+    for(i = 0 ; i <= 2 ; i++)
+    {
+        led_param[i]='\0';
+    }
+
+    pch = strtok (pArg, ",");
+    led_param[0]=pch;
+
+    while (pch != NULL && params <= 2)
+    {
+        led_param[params] = pch;
+        params++;
+        pch = strtok(NULL, ",");
+    }
+
+    if (params == 1)
+    {
+        if (led_param[0][0] == '-' && (led_param[0][1] == 'h' || led_param[0][1] == 'H' || led_param[0][1] == '?'))
+        {
+            return print_led_help(NULL);
+        }
+    }
+
+    if (led_param[0] != NULL)
+    {
+        if (led_param[0][0] == '-' && (led_param[0][1] == 'h' || led_param[0][1] == 'H' || led_param[0][1] == '?'))
+        {
+            return print_led_help(NULL);
+        }
+        else if (!isDigit((int)led_param[0][0]))
+        {
+            return print_led_help("Please provide LED Number" NEWLINE);
+        }
+        led_num = atoi(&led_param[0][0]);
+    }
+    else
+    {
+        return print_led_help("Please provide LED Number" NEWLINE);
+    }
+
+    if (led_param[1] != NULL)
+    {
+        if (!isDigit((int)led_param[1][0])) {
+            return print_led_help("Please provide LED State Number" NEWLINE);
+        }
+        led_state = atoi(&led_param[1][0]);
+    }
+    else
+    {
+        return print_led_help("Please provide LED State Number" NEWLINE);
+    }
+
+    switch (led_state)
+    {
+        case 1:
+            new_led_state = LED_STATE_HOLD;
+            break;
+        case 2:
+            new_led_state = LED_STATE_OFF;
+            break;
+        case 3:
+            new_led_state = LED_STATE_BLINK_FAST;
+            break;
+        case 4:
+            new_led_state = LED_STATE_BLINK_SLOW;
+            break;
+    }
+
+    switch (led_num)
+    {
+        case 1:
+            LED_SetBlue(new_led_state);
+            break;
+        case 2:
+            LED_SetGreen(new_led_state);
+            break;
+        case 3:
+            LED_SetYellow(new_led_state);
+            break;
+        case 4:
+            LED_SetRed(new_led_state);
+            break;
+    }
+}
+
+static void get_set_dps_idscope(char *pArg)
+{
+    char *dps_param;
+    uint8_t i;
```

Comment:
nit: keep this closer to where it is used (line 505)

Reply:
done